jvector-base/src/main/java/io/github/jbellis/jvector/pq/BinaryQuantization.java

```diff
@@ -63,12 +63,18 @@ public CompressedVectors createCompressedVectors(Object[] compressedVectors) {
     }
 
     @Override
-    public long[][] encodeAll(RandomAccessVectorValues ravv, ForkJoinPool simdExecutor) {
-        return simdExecutor.submit(() -> IntStream.range(0, ravv.size())
+    public CompressedVectors encodeAll(RandomAccessVectorValues ravv, ForkJoinPool simdExecutor) {
+        var cv = simdExecutor.submit(() -> IntStream.range(0, ravv.size())
                 .parallel()
```

Comment:
out of curiosity:
does it really help to use "parallel" here ?

Reply:
Massively.